File: bitlap-core/src/main/kotlin/org/bitlap/core/storage/BitlapWriter.kt
/**
 * Copyright (C) 2023 bitlap.org .
 */
package org.bitlap.core.storage

import java.io.Closeable

/**
 * Desc: mdm bitlap writer
 */
interface BitlapWriter<T> : Closeable {

    fun writeBatch(rows: Iterable<T>)

    fun write(row: T)
}

File: bitlap-core/src/main/scala/org/bitlap/core/storage/BitlapWriter.scala
/**
 * Copyright (C) 2023 bitlap.org .
 */
package org.bitlap.core.storage

import java.io.Closeable

/** mdm bitlap writer
 */
trait BitlapWriter[T] extends Closeable {

  def writeBatch(rows: Iterable[T]): Unit

  def write(row: T): Unit
}
